Add easier discoverability and clarity to the MDC appender docs (#8638)
open-telemetry/opentelemetry-java-instrumentation: docs/logger-mdc-instrumentation.md

  (same as `Span.current().getSpanContext().getTraceFlags().asHex()`).

Those three pieces of information can be included in log statements produced by the logging library
by specifying them in the pattern/format. This way any services or tools that parse the application 
logs can correlate traces/spans with log statements.

> > Note: If the current `Span` is invalid, there will be no OTEL info in the MDC

Reviewer: Nit: slightly rephrased
```suggestion
> Note: If the current `Span` is invalid, the OpenTelemetry appender will not inject any trace information.
```

## Supported logging libraries

> > Note: There are also log appenders for exporting logs to OpenTelemetry, not to be confused with the MDC appenders

Reviewer: 👍 
```suggestion
> Note: There are also log appenders for exporting logs to OpenTelemetry, not to be confused with the MDC appenders. 
```

> | Library | Version | Documentation                                                                        |

Reviewer: WDYT about using the same column names as in https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/docs/supported-libraries.md#libraries--frameworks ?
```suggestion
| Library | Auto-instrumented versions | Standalone Library Instrumentation                                                                        |
```
